aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -138,101 +199,142 @@ pub(crate) fn is_valid_txn_arg<S: MoveResolverExt>(
     }
 }
 
-// Validate arguments. Walk through the arguments and according to the signature
-// validate arguments that require so.
+// Construct arguments. Walk through the arguments and according to the signature
+// construct arguments that require so.
 // TODO: This needs a more solid story and a tighter integration with the VM.
-// Validation at the moment is only for Strings and Vector of them, so we
-// manually walk the serialized args until we find a string.
-// This is obviously brittle and something to change at some point soon.
-pub(crate) fn validate_args<S: MoveResolverExt>(
-    session: &SessionExt<S>,
+pub(crate) fn construct_args<S: MoveResolverExt>(
+    session: &mut SessionExt<S>,
     idxs: &[usize],
-    args: &[Vec<u8>],
+    args: &mut [Vec<u8>],
```

Comment:
How about returning the new args instead? Personally I found the overwriting confusing and unintuitive, as it's hard for me as a reader to realize it's changing the underlying data from one format into another the first time I see it.

Reply:
+1

Reply:
I think this is a pure taste things. Mutability is clearly idiomatic in Rust. I did not find this difficult to read.